tensorflow/compiler/mlir/hlo/include/mlir-hlo/Dialect/mhlo/IR/hlo_ops.td

```diff
@@ -2162,4 +2162,124 @@ def HLO_ReducePrecisionOp :
   let results = (outs HLO_FpTensor:$output);
 }
 
+def HLO_H2DOp: HLO_Op<"h2d", [NoSideEffect, SameOperandsAndResultType]> {
+  let summary = "H2D operator";
+  let description = [{
+    Copy `operand` from host to device.
+  }];
+  let arguments = (ins HLO_Tensor);
+  let results = (outs HLO_Tensor);
+  let hasCustomHLOConverter = 1;
+}
+
+def HLO_D2HOp: HLO_Op<"d2h", [NoSideEffect, SameOperandsAndResultType]> {
+  let summary = "D2H operator";
+  let description = [{
+    Copy `operand` from device to host.
+  }];
+  let arguments = (ins HLO_Tensor);
+  let results = (outs HLO_Tensor);
+  let hasCustomHLOConverter = 1;
+}
+
+def HLO_RealDynamicSliceOp: HLO_Op<
+      "dynamic_slice",
+      [NoSideEffect, AllElementTypesMatch<["operand", "result"]>,
+       AllTypesMatch<["start_indices", "limit_indices", "strides"]>]> {
+  let summary = "Real Dynamic Slice operator";
+  let description = [{
+    The dynamic shape version of SliceOp. Extracts a sub-array from the input 
+    array according to start_indices, limit_indices and strides. Expect 
+    start_indices/limit_indices/strides to be statically shaped and matching 
+    the rank of the input.
+  }];
+  let arguments = (ins
+    HLO_Tensor:$operand,
+    HLO_Tensor:$start_indices,
+    HLO_Tensor:$limit_indices,
+    HLO_Tensor:$strides
```

Comment:
Instead of HLO_Tensor for these, can you use `HLO_DimensionTensor`? If it isn't the right one, can we add another one that would be suitable to implement the individual constraint on these operands?

Reply:
Done. We change these arguments' type to HLO_DimensionTensor.